node/src/crypto/asymmetric_key.rs

```diff
@@ -64,25 +61,10 @@ pub fn verify<T: AsRef<[u8]>>(
             "signatures based on the system key cannot be verified",
         ))),
         (Signature::Ed25519(signature), PublicKey::Ed25519(public_key)) => public_key
-            .verify_strict(
-                message.as_ref(),
-                &ed25519_dalek::Signature::from_bytes(signature).map_err(|_| {
-                    Error::AsymmetricKey(format!(
-                        "failed to construct Ed25519 signature from {:?}",
-                        &signature[..]
-                    ))
-                })?,
-            )
+            .verify_strict(message.as_ref(), signature)
             .map_err(|_| Error::AsymmetricKey(String::from("failed to verify Ed25519 signature"))),
```

Comment:
Can we add an error variant here instead of using this string?

Reply:
I figure that work would be part of the follow up PR

Reply:
I agree.  We shouldn't have error variants with strings since they are security hazards, so we'll clean this up when we do remove those.